api/proto/teleport/legacy/client/proto/joinservice.proto

```diff
@@ -194,6 +194,116 @@ message RegisterUsingOracleMethodResponse {
   }
 }
 
+// RegisterUsingBoundKeypairInitialRequest is the initial request sent during
+// the join process for bound-keypair joining.
+message RegisterUsingBoundKeypairInitialRequest {
+  // Registration parameters common to all join methods.
+  types.RegisterUsingTokenRequest join_request = 1;
+
+  // If set, attempts to bind a new keypair using an initial join secret.
+  // Any value set here will be ignored if a keypair is already bound.
+  string initial_join_secret = 2;
+
+  // If set, requests a rotation to the new public key. The joining challenge
+  // must first be completed using the previous key, and upon completion a new
+  // challenge will be issued for this key. Certificates will only be returned
+  // after the second challenge is complete. On initial join when using an
+  // initial join secret, this value should be set to the desired public key.
+  // This key is written in SSH public key format.
+  string new_public_key = 3;
```

Comment:
Design question, my initial design was missing a way to inform the client that a rotation had actually been requested server-side. I've now added an explicit ceremony phase for this (`RegisterUsingBoundKeypairRotationRequest`) so the server will explicitly ask the client to provide a new public key, and then issue the 2nd challenge against it.

Anyway, with that mechanism in place, I think we could technically remove this field. Do we want to keep it and allow clients to initiate a key rotation on their own? I'm sort of leaning toward "no", now that we have the choice, but am curious for others' thoughts.

Reply:
I'd be in favor of removing it to simplify things, unless there's a compelling reason to allow clients to initiate the rotation?

Reply:
I agree, and think I'll remove it. I'd rather have one flow now, and we can look into cleaner ways to implement client-driven rotation if that ever comes up. I don't think I like the complexity it adds to the challenge ceremony if we have to account for both options.

Reply:
I've removed the field now, so rotation will now be exclusively driven by the server.